### DISCUSSION TITLE
Interaction data not showing

### DISCUSSION THREAD

**Discussion by aschutz32** [jspsych/jsPsych#2962] [Q&A]
Hello! I appreciate the help lately with getting my experimental code working. I am now trying to track specifically when participants exit fullscreen via 'on_interaction_data_update' within the initJsPsych call. This seems rather simple to do, however it is not showing within the data regardless if I exit fullscreen or switch tabs. I am using version 7.3.1

_Do I have it written incorrectly?_ 

I have looked through other posts and nothing I have tried appears to work. Any suggestions are more than welcome! Thank you!

var jsPsych = initJsPsych({
      on_interaction_data_update: function(data) {
        console.log(JSON.stringify(data));
      },
      on_finish: function(){
        jsPsych.data.displayData();
      }
    });

**Comment by jodeleeuw** [comment]
Hi @aschutz32,

I wrote up a test page to verify that the interaction data is being recorded correctly:

```html
<!DOCTYPE html>
<html>
  <head>
    <script src="https://unpkg.com/jspsych@7.3.1"></script>
    <script src="https://unpkg.com/@jspsych/plugin-fullscreen"></script>
    <script src="https://unpkg.com/@jspsych/plugin-html-button-response"></script>
    <link rel="stylesheet" href="https://unpkg.com/jspsych@7.3.1/css/jspsych.css">
  </head>
  <body></body>
  <script>
    const jsPsych = initJsPsych({
      on_interaction_data_update: (data) => {
        console.log(data)
      }
    })

    const timeline = [];

    timeline.push({
      type: jsPsychFullscreen,
      fullscreen_mode: true
    });

    timeline.push({
      type: jsPsychHtmlButtonResponse,
      stimulus: 'Press the button to continue.',
      choices: ['Continue']
    });

    jsPsych.run(timeline)
  </script>
</html>
```

This works for me, and the interaction data shows up in the `console.log()`:

![image](https://user-images.githubusercontent.com/595524/222528851-fe98bb02-fc7f-467d-9d6c-1aaee004864c.png)

If your question is about the data that is shown with `jsPsych.data.displayData()`, that data does not include the interaction data. The interaction data is stored separately, and can be retrieved with `jsPsych.data.getInteractionData()` ([docs](https://www.jspsych.org/7.3/reference/jspsych-data/#jspsychdatagetinteractiondata)).
